### DISCUSSION TITLE
Serializing Compiled Schemas

### DISCUSSION THREAD

**Discussion by Ghirigoro** [sinclairzx81/typebox#769] [Q&A] [answered]
My understanding is that schemas can't reliably be serialized/deserialized due to containing additional type information in [Kind] and other fields. But, is it possible to serialize/deserialize the schemas that are compiled by `TypeCompiler`? Looking at the output there's a `code` field that could be `eval`ed at runtime. On first glance it looks like it would work. Is this a dumb idea? Is there a better way to add/remove schemas dynamically at runtime? (My ultimate goal is to store them in a database so that new data types can be added and validated without having to rebuild and redeploy the server).

**Comment by robert-e-davidson3** [comment]
Is there docs or something about how typebox cannot reliably (de)serialize its schemas? That's the exact usecase I chose typebox for, unfortunately.

**Reply by sinclairzx81** [reply, marked as answer]
Typebox 1.0 (and previous versions) support full serialization. TypeBox 1.0 added support for deserialization by consequence of having validators able to validate raw Json Schema. You can transmit TypeBox types over Http and have remote systems load and validate with them.

**Reply by robert-e-davidson3** [reply]
Oh! Fantastic news. I couldn't get it to work with an older version but I'll upgrade to 1.0 and try again. Thanks!
